### PR TITLE
feat: add lib and es6 bundles, use closure compiler, remove gulp

### PR DESCRIPTION
**Pull request by frederickfogerty**
## Description

This PR improves the build step of this library. The minified JS build is reduced from `18K -> 11K = 40% smaller`. The css size is unchanged.

There have been some changes to the source code to support the closure compiler. There might be a chance that things that worked before may not work now due to the optimisations it performs. I have tried to cover all common usage, but there may be some edge cases which I have not tested. All the tests still pass, and I've ensured that the tests test the compiled code.

Gulp has also been removed in favour of npm scripts.

### New Feature

- [x] If this is a big feature with breaking changes, consider [opening an issue][issues] to discuss first. This is completely up to you, but please keep in mind that your PR might not be accepted.
- [x] Run unit tests to ensure all existing tests are still passing
- [x] Add new passing unit tests to cover the code introduced by your PR
- [x] Update the readme
- [x] Add some [steps](#steps-to-test) so we can test your cool new feature!
- [x] The PR title is in the [conventional commit](#conventional-commit-spec) format: e.g. `feat(<area>): added new way to do this cool thing #issue-number`
- [x] Add your info to the [contributors](#packagejson-contributors) array in package.json!

## Steps to Test

1. Clone this PR and run `npm install` to update dependencies
2. Run `npm run build` and open `index.html` in a browser, and test that Drift still works normally
3. Create a consuming package, npm link, and ensure that Drift works when importing using `import Drift from 'drift-zoom'`

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const webpack = require("webpack");
+const path = require("path");
+const env = require("yargs").argv.env; // use --env with webpack 2
+const pkg = require("./package.json");
+const UglifyJsPlugin = require("uglifyjs-webpack-plugin");
```
Comment (jayeb) [inline webpack.config.js]: This doesn't appear to be used in this file.

File: bower.json [hunk 2]
```diff
@@ -25,7 +25,6 @@
     "src",
     "test",
     "CONTRIBUTING.md",
-    "gulpfile.js",
     "index.html",
     "karmaConfig.js",
```
Comment (jayeb) [inline bower.json]: Some of the files in this list need updating

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -158,13 +157,13 @@ npm install
 Once you're configured, building the JavaScript from the command line is easy:
 
 ```
-gulp            # build Drift from source
-gulp watch      # watch for changes and build automatically, testing with PhantomJS
-gulp test-local # run the test against local browsers only (Chrome, Safari, Firefox)
-gulp test-full  # run the test suite against all supported browsers (using SauceLabs)
+npm run build			# build Drift from source
+npm run build:watch     # watch for changes and build automatically
```
Comment (jayeb) [inline CONTRIBUTING.md]: [nit] Handsome tab spacing is broken here

**Comment by fleck** [comment]
👏This looks awesome! Also, that run-p package seems really nice. Looks as though it would make working with scripts much easier.
